index.js

```diff
@@ -725,6 +725,198 @@ PersonaClient.prototype._removeTokenFromCache = function (id, secret, callback)
 };
 
 /**
+ * Get scope information for a user
+ * @param guid
+ * @param token
+ * @param callback
+ */
+PersonaClient.prototype.getScopesForUser = function(guid, token, callback) {
+    try {
+        _.map([guid, token], function (arg) {
+            if (!_.isString(arg)) {
+                throw "guid and token are required strings";
+            }
+        });
+    } catch (e) {
+        callback(e,null);
+        return;
+    }
+
+    var _this = this;
+
+    var options = {
+            hostname: _this.config.persona_host,
+            port: _this.config.persona_port,
+            path: "/1/clients/" + guid,
+            method: 'GET',
+            headers: {
+                Authorization: "Bearer " + token,
+            },
+        },
+        req = _this.http.request(options, function (resp) {
+            if (resp.statusCode === 200) {
+                var str = '';
+
+                resp.on("data", function (chunk) {
+                    str += chunk;
+                });
+
+                resp.on("end", function () {
+                    var data;
+                    try {
+                        data = JSON.parse(str);
+                    } catch (e) {
+                        callback("Error parsing response from persona: " + str, null);
+                        return;
+                    }
+
+                    if (data) {
+                        if (data.error) {
+                            callback(data.error, null);
+                        } else {
+                            if (data.scope) {
+                                callback(null, data.scope);
+                            } else {
+                                callback("Scopes not found for Guid", null);
+                            }
+                        }
+                    } else {
+                        callback("Could not get Scopes for Guid", null);
+                    }
+                });
+            } else {
+                var err = "getScopesForUser failed with status code " + resp.statusCode;
+                _this.error(err);
+                callback(err, null);
+            }
+        });
+
+    req.on("error", function (e) {
+        var err = "getScopesForUser problem: " + e.message;
+        _this.error(err);
+        callback(err, null);
+    });
+
+    req.end();
+};
+
+/**
+ * Helper method to set the scopes for a user - can add or remove a scope by passing scopeChange appropriately
+ * @param guid
+ * @param token
+ * @param scopeChange
+ * @param callback
+ */
+PersonaClient.prototype._setScopesForUser = function(guid, token, scopeChange, callback) {
+    try {
+        _.map([guid, token], function (arg) {
+            if (!_.isString(arg)) {
+                throw "guid, token are required strings";
```

Comment:
In fact, you are already guarding for this in the public callers. This is complete defensive overkill.